Use Jinja sandboxed environment

amanda11 opened the pull request:
Use Jinja sandboxed environment, so that any templates that attempt to access insecure code will be rejected.

m4dcoder reviewed (changes requested):
- Do we need to update the Jinja version to >= 2.11 in requirements.txt?
- Can you also update description of this PR on why we are using sandboxed environment
- Please add a changelog entry.
- The CI weren't building because workflows was automatically disabled by Github due to inactivity. I believe there's a build failure that need to be fixed.ding because workflows was automatically disabled by Github due to inactivity. I believe there's an unrelated build failure that need to be fixed.

amanda11 commented:
> * Do we need to update the Jinja version to >= 2.11 in requirements.txt?
> * Can you also update description of this PR on why we are using sandboxed environment
> * Please add a changelog entry.
> * The CI weren't building because workflows was automatically disabled by Github due to inactivity. I believe there's a build failure that need to be fixed.ding because workflows was automatically disabled by Github due to inactivity. It's resolved now.  But I believe there's an unrelated build failure that need to be fixed.

@m4dcoder Addressed your review comments (stealing the sphinx change from your PR!)

m4dcoder reviewed (approved):
Thank you!
